runtime/misc.c

```diff
@@ -202,3 +202,18 @@ int caml_runtime_warnings_active(void)
   }
   return 1;
 }
+
+uintnat caml_float_array_alloc_warning = 0;
+static int caml_float_array_alloc_warning_first = 1;
+uintnat caml_float_array_alloc_callstack_size = 10;
```

Comment:
You can merge `caml_float_array_alloc_warning` and `caml_float_array_alloc_callstack_size` in a single variable, where a value of `0` means "disabled" and anything "> 0" means enabled. That way you do not have to give a default value to `caml_float_array_alloc_callstack_size` and also you get to use the value passed in `OCAMLRUNPARAM`.

Reply:
What should be the API for `Sys` in this case ? Just one argument with the "size" of the callstack ?

Reply:
I think the `Sys` API is fine as it is.

Reply:
I mean we have to merge two arguments into one then.

Reply:
Merging both variables is more of an implementation detail. Semantically the data of whether the warning is enabled/disabled and the length of the backtrace are different. So the OCaml API can stay as it is, where the pair of arguments `(true, n)` maps to `n` on the C side and `(false, n)` maps to `0` on the C side. For extra safety the argument on the OCaml side may be normalized so that it is always `>= 1`.